src/main/java/org/example/myWork/logic/TaskDaoImpl.java

```diff
@@ -18,22 +19,25 @@
 
     @Override
     public List<Task> findAllFiltered(String query, boolean excludeCompleted, User user) {
-        StringBuilder jpql = new StringBuilder("from Task t ");
+        StringBuilder jpql = new StringBuilder("SELECT t FROM Task t ");
         List<String> conditions = new ArrayList<>(2);
-        TypedQuery<Task> typedQuery = entityManager.createQuery(jpql.toString(), Task.class);
+        List<Pair<String, Object>> parameters = new ArrayList<>(2);
         if (Strings.isNotBlank(query)) {
             conditions.add("t.description like :desc");
-            typedQuery.setParameter("desc", "%" + query + "%");
+            parameters.add(new Pair<String, Object>("desc", "%" + query + "%"));
         }
         if (excludeCompleted) {
             conditions.add("t.done <> true");
         }
         if (user != null) {
-            conditions.add("t.owner like :");
-            typedQuery.setParameter("own", "%" + user + "%");
+            conditions.add("t.owner.id = " + user.getId());
```

Comment:
В целом, не критичное замечание, но я бы тогда уж привел к единому виду, типа 
```suggestion
            conditions.add("t.owner.id = :ownerId");
            parameters.add(new Pair<String, Object>("ownerId", user.getId()));
```

Reply:
учел, поправил